Regression in version 1.11.6 . Generated test fails with ` ValueError: Invalid request.`

Starting in version 1.11.6, generated unit tests for `google/streetview/publish/v1` fail with ` ValueError: Invalid request.`

I've narrowed the problematic commit to https://github.com/googleapis/gapic-generator-python/commit/812abceeb68d86f3902974aaeef8a58cfd6e7515

See snippet of stack trace below
```
bazel run //google/streetview/publish/v1:publish_py_gapic_test 
...
_________________________ test_update_photo_rest[dict] _________________________

request_type = <class 'dict'>

    @pytest.mark.parametrize("request_type", [
        rpcmessages.UpdatePhotoRequest,
        dict,
    ])
    def test_update_photo_rest(request_type):
        client = StreetViewPublishServiceClient(
            credentials=ga_credentials.AnonymousCredentials(),
            transport="rest",
        )
    
        # send a request that will satisfy transcoding
        request_init = {'photo': {'photo_id': {'id': 'sample1'}}}
        request_init["photo"] = {'photo_id': {'id': 'sample1'}, 'upload_reference': {'upload_url': 'upload_url_value'}, 'download_url': 'download_url_value', 'thumbnail_url': 'thumbnail_url_value', 'share_link': 'share_link_value', 'pose': {'lat_lng_pair': {'latitude': 0.86, 'longitude': 0.971}, 'altitude': 0.86, 'heading': 0.72, 'pitch': 0.536, 'roll': 0.441, 'gps_record_timestamp_unix_epoch': {'seconds': 751, 'nanos': 543}, 'level': {'number': 0.649, 'name': 'name_value'}, 'accuracy_meters': 0.15940000000000001}, 'connections': [{'target': {}}], 'capture_time': {}, 'upload_time': {}, 'places': [{'place_id': 'place_id_value', 'name': 'name_value', 'language_code': 'language_code_value'}], 'view_count': 1091, 'transfer_status': 1, 'maps_publish_status': 1}
        # The version of a generated dependency at test runtime may differ from the version used during generation.
        # Delete any fields which are not present in the current runtime dependency
        # See https://github.com/googleapis/gapic-generator-python/issues/1748
    
        # Determine if the message type is proto-plus or protobuf
        is_message_proto_plus_type = not hasattr(rpcmessages.UpdatePhotoRequest.meta.fields["photo"].message, "DESCRIPTOR")
    
        if is_message_proto_plus_type:
            message_fields = rpcmessages.UpdatePhotoRequest.meta.fields["photo"].message.meta.fields
        else:
            message_fields = rpcmessages.UpdatePhotoRequest.meta.fields["photo"].message.DESCRIPTOR.fields
    
        subfields_not_in_runtime = []
    
        # Get all subfields for the message
        runtime_nested_fields = [
            (field.name, subfield.name)
            for field in message_fields
            if hasattr(field, "message_type") and field.message_type
            for subfield in field.message_type.fields
        ]
    
        # For each item in the sample request, create a list of sub fields which are not present at runtime
        for field, value in request_init["photo"].items():
            result = None
            is_repeated = False
            # For repeated fields
            if isinstance(value, list) and len(value):
                is_repeated = True
                result = value[0]
            # For fields where the type is another message
            if isinstance(value, dict):
                result = value
    
            if result and hasattr(result, "keys"):
                for subfield in result.keys():
                    if (field, subfield) not in runtime_nested_fields:
                        subfields_not_in_runtime.append(
                            {"field": field, "subfield": subfield, "is_repeated": is_repeated}
                        )
    
        # Remove fields from the sample request which are not present in the runtime version of the dependency
        for subfield_to_delete in subfields_not_in_runtime:
            field = subfield_to_delete.get("field")
            field_repeated = subfield_to_delete.get("is_repeated")
            subfield = subfield_to_delete.get("subfield")
            if subfield:
                if field_repeated:
                    for i in range(0, len(request_init["photo"][field])):
                        del request_init["photo"][field][i][subfield]
                else:
                    del request_init["photo"][field][subfield]
        request = request_type(**request_init)
    
        # Mock the http request call within the method and fake a response.
        with mock.patch.object(type(client.transport._session), 'request') as req:
            # Designate an appropriate value for the returned response.
            return_value = resources.Photo(
                  download_url='download_url_value',
                  thumbnail_url='thumbnail_url_value',
                  share_link='share_link_value',
                  view_count=1091,
                  transfer_status=resources.Photo.TransferStatus.NEVER_TRANSFERRED,
                  maps_publish_status=resources.Photo.MapsPublishStatus.PUBLISHED,
            )
    
            # Wrap the value into a proper Response obj
            response_value = Response()
            response_value.status_code = 200
            # Convert return value to protobuf type
            return_value = resources.Photo.pb(return_value)
            json_return_value = json_format.MessageToJson(return_value)
    
            response_value._content = json_return_value.encode('UTF-8')
            req.return_value = response_value
>           response = client.update_photo(request)

google/streetview/publish/v1/publish_py_gapic_srcjar.py/tests/unit/gapic/publish_v1/test_street_view_publish_service.py:4659: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
google/streetview/publish/v1/publish_py_gapic_srcjar.py/google/streetview/publish_v1/services/street_view_publish_service/client.py:1150: in update_photo
    response = rpc(
../gapic_generator_python_pip_deps_google_api_core/site-packages/google/api_core/gapic_v1/method.py:131: in __call__
    return wrapped_func(*args, **kwargs)
../gapic_generator_python_pip_deps_google_api_core/site-packages/google/api_core/retry.py:366: in retry_wrapped_func
    return retry_target(
../gapic_generator_python_pip_deps_google_api_core/site-packages/google/api_core/retry.py:204: in retry_target
    return target()
../gapic_generator_python_pip_deps_google_api_core/site-packages/google/api_core/timeout.py:120: in func_with_timeout
    return func(*args, **kwargs)
../gapic_generator_python_pip_deps_google_api_core/site-packages/google/api_core/grpc_helpers.py:75: in error_remapped_callable
    return callable_(*args, **kwargs)
google/streetview/publish/v1/publish_py_gapic_srcjar.py/google/streetview/publish_v1/services/street_view_publish_service/transports/rest.py:1678: in __call__
    transcoded_request = path_template.transcode(http_options, pb_request)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

http_options = [{'body': 'photo', 'method': 'put', 'uri': '/v1/photo/{photo.photo_id.id}'}]
message = photo {
  photo_id {
  }
  upload_reference {
  }
  download_url: "download_url_value"
  thumbnail_url: "thumbnail_url...d_time {
  }
  places {
  }
  view_count: 1091
  transfer_status: NEVER_TRANSFERRED
  maps_publish_status: PUBLISHED
}

request_kwargs = {}
bindings = [('/v1/photo/{photo.photo_id.id}', [('photo.photo_id.id', None)])]
http_option = {'body': 'photo', 'method': 'put', 'uri': '/v1/photo/{photo.photo_id.id}'}
request = {'uri': '/v1/photo/'}, uri_template = '/v1/photo/{photo.photo_id.id}'
fields = [('photo.photo_id.id', None)], path_args = {'photo.photo_id.id': ''}

    def transcode(http_options, message=None, **request_kwargs):
        """Transcodes a grpc request pattern into a proper HTTP request following the rules outlined here,
        https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L44-L312
    
         Args:
             http_options (list(dict)): A list of dicts which consist of these keys,
                 'method'    (str): The http method
                 'uri'       (str): The path template
                 'body'      (str): The body field name (optional)
                 (This is a simplified representation of the proto option `google.api.http`)
    
             message (Message) : A request object (optional)
             request_kwargs (dict) : A dict representing the request object
    
         Returns:
             dict: The transcoded request with these keys,
                 'method'        (str)   : The http method
                 'uri'           (str)   : The expanded uri
                 'body'          (dict | Message)  : A dict or a Message representing the body (optional)
                 'query_params'  (dict | Message)  : A dict or Message mapping query parameter variables and values
    
         Raises:
             ValueError: If the request does not match the given template.
        """
        transcoded_value = message or request_kwargs
        bindings = []
        for http_option in http_options:
            request = {}
    
            # Assign path
            uri_template = http_option["uri"]
            fields = [
                (m.group("name"), m.group("template"))
                for m in _VARIABLE_RE.finditer(uri_template)
            ]
            bindings.append((uri_template, fields))
    
            path_args = {field: get_field(transcoded_value, field) for field, _ in fields}
            request["uri"] = expand(uri_template, **path_args)
    
            if not validate(uri_template, request["uri"]) or not all(path_args.values()):
                continue
    
            # Remove fields used in uri path from request
            leftovers = copy.deepcopy(transcoded_value)
            for path_field, _ in fields:
                delete_field(leftovers, path_field)
    
            # Assign body and query params
            body = http_option.get("body")
    
            if body:
                if body == "*":
                    request["body"] = leftovers
                    if message:
                        request["query_params"] = message.__class__()
                    else:
                        request["query_params"] = {}
                else:
                    try:
                        if message:
                            request["body"] = getattr(leftovers, body)
                            delete_field(leftovers, body)
                        else:
                            request["body"] = leftovers.pop(body)
                    except (KeyError, AttributeError):
                        continue
                    request["query_params"] = leftovers
            else:
                request["query_params"] = leftovers
            request["method"] = http_option["method"]
            return request
    
        bindings_description = [
            '\n\tURI: "{}"'
            "\n\tRequired request fields:\n\t\t{}".format(
                uri,
                "\n\t\t".join(
                    [
                        'field: "{}", pattern: "{}"'.format(n, p if p else "*")
                        for n, p in fields
                    ]
                ),
            )
            for uri, fields in bindings
        ]
    
>       raise ValueError(
            "Invalid request."
            "\nSome of the fields of the request message are either not initialized or "
            "initialized with an invalid value."
            "\nPlease make sure your request matches at least one accepted HTTP binding."
            "\nTo match a binding the request message must have all the required fields "
            "initialized with values matching their patterns as listed below:{}".format(
                "\n".join(bindings_description)
            )
        )
E       ValueError: Invalid request.
E       Some of the fields of the request message are either not initialized or initialized with an invalid value.
E       Please make sure your request matches at least one accepted HTTP binding.
E       To match a binding the request message must have all the required fields initialized with values matching their patterns as listed below:
E               URI: "/v1/photo/{photo.photo_id.id}"
E               Required request fields:
E                       field: "photo.photo_id.id", pattern: "*"

../gapic_generator_python_pip_deps_google_api_core/site-packages/google/api_core/path_template.py:337: ValueError
=========================== short test summary info ============================
FAILED google/streetview/publish/v1/publish_py_gapic_srcjar.py/tests/unit/gapic/publish_v1/test_street_view_publish_service.py::test_update_photo_rest[UpdatePhotoRequest]
FAILED google/streetview/publish/v1/publish_py_gapic_srcjar.py/tests/unit/gapic/publish_v1/test_street_view_publish_service.py::test_update_photo_rest[dict]
2 failed, 341 passed, 8 warnings in 5.42s
```

--- a/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -1019,22 +1019,30 @@ def test_{{ method.name|snake_case }}_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr({{ method.input.ident }}.meta.fields["{{ field.name }}"].message, "DESCRIPTOR")
+    test_field = {{ method.input.ident }}.meta.fields["{{ field.name }}"]
 
-    if is_message_proto_plus_type:
-        message_fields = {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.meta.fields
-    else:
-        message_fields = {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["{{ field.name }}"].items():

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
@@ -892,22 +892,30 @@ def test_{{ method_name }}_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr({{ method.input.ident }}.meta.fields["{{ field.name }}"].message, "DESCRIPTOR")
+    test_field = {{ method.input.ident }}.meta.fields["{{ field.name }}"]
 
-    if is_message_proto_plus_type:
-        message_fields = {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.meta.fields
-    else:
-        message_fields = {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["{{ field.name }}"].items():

--- a/tests/fragments/test_google_protobuf_type.proto
+++ b/tests/fragments/test_google_protobuf_type.proto
@@ -34,20 +34,27 @@ service MyServiceWithProtobufType {
   rpc MyMethod(MethodRequestWithProtobufType)
       returns (MethodResponseWithProtobufType) {
     option (google.api.http) = {
-      post: "/v1/services/{service_name}/configs"
+      post: "/v1/services/{service_name}/configs/{test_message.another_message.another_field}"
       body: "test_message"
     };
-    option (google.api.method_signature) = "service_name,test_message";
+    option (google.api.method_signature) = "service_name,test_message,another_string";
   }
 }
 
 message MethodRequestWithProtobufType {
   string service_name = 1 [(google.api.field_behavior) = REQUIRED];
   TestMessage test_message = 2 [(google.api.field_behavior) = REQUIRED];
+  string another_string = 3 [(google.api.field_behavior) = REQUIRED];
 }
 
 message TestMessage {
+  string name = 1 [(google.api.field_behavior) = REQUIRED];
   repeated google.protobuf.Type types = 2 [(google.api.field_behavior) = REQUIRED];
+  AnotherTestMessage another_message = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+message AnotherTestMessage {
+  string another_field = 1;
 }
 
 message MethodResponseWithProtobufType {

--- a/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py
+++ b/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py
@@ -9998,22 +9998,30 @@ def test_create_saved_query_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(asset_service.CreateSavedQueryRequest.meta.fields["saved_query"].message, "DESCRIPTOR")
+    test_field = asset_service.CreateSavedQueryRequest.meta.fields["saved_query"]
 
-    if is_message_proto_plus_type:
-        message_fields = asset_service.CreateSavedQueryRequest.meta.fields["saved_query"].message.meta.fields
-    else:
-        message_fields = asset_service.CreateSavedQueryRequest.meta.fields["saved_query"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["saved_query"].items():
@@ -10843,22 +10851,30 @@ def test_update_saved_query_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(asset_service.UpdateSavedQueryRequest.meta.fields["saved_query"].message, "DESCRIPTOR")
+    test_field = asset_service.UpdateSavedQueryRequest.meta.fields["saved_query"]
 
-    if is_message_proto_plus_type:
-        message_fields = asset_service.UpdateSavedQueryRequest.meta.fields["saved_query"].message.meta.fields
-    else:
-        message_fields = asset_service.UpdateSavedQueryRequest.meta.fields["saved_query"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["saved_query"].items():

--- a/tests/integration/goldens/eventarc/tests/unit/gapic/eventarc_v1/test_eventarc.py
+++ b/tests/integration/goldens/eventarc/tests/unit/gapic/eventarc_v1/test_eventarc.py
@@ -6230,22 +6230,30 @@ def test_create_trigger_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(eventarc.CreateTriggerRequest.meta.fields["trigger"].message, "DESCRIPTOR")
+    test_field = eventarc.CreateTriggerRequest.meta.fields["trigger"]
 
-    if is_message_proto_plus_type:
-        message_fields = eventarc.CreateTriggerRequest.meta.fields["trigger"].message.meta.fields
-    else:
-        message_fields = eventarc.CreateTriggerRequest.meta.fields["trigger"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["trigger"].items():
@@ -6538,22 +6546,30 @@ def test_update_trigger_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(eventarc.UpdateTriggerRequest.meta.fields["trigger"].message, "DESCRIPTOR")
+    test_field = eventarc.UpdateTriggerRequest.meta.fields["trigger"]
 
-    if is_message_proto_plus_type:
-        message_fields = eventarc.UpdateTriggerRequest.meta.fields["trigger"].message.meta.fields
-    else:
-        message_fields = eventarc.UpdateTriggerRequest.meta.fields["trigger"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["trigger"].items():
@@ -7612,22 +7628,30 @@ def test_create_channel_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(eventarc.CreateChannelRequest.meta.fields["channel"].message, "DESCRIPTOR")
+    test_field = eventarc.CreateChannelRequest.meta.fields["channel"]
 
-    if is_message_proto_plus_type:
-        message_fields = eventarc.CreateChannelRequest.meta.fields["channel"].message.meta.fields
-    else:
-        message_fields = eventarc.CreateChannelRequest.meta.fields["channel"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["channel"].items():
@@ -7920,22 +7944,30 @@ def test_update_channel_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(eventarc.UpdateChannelRequest.meta.fields["channel"].message, "DESCRIPTOR")
+    test_field = eventarc.UpdateChannelRequest.meta.fields["channel"]
 
-    if is_message_proto_plus_type:
-        message_fields = eventarc.UpdateChannelRequest.meta.fields["channel"].message.meta.fields
-    else:
-        message_fields = eventarc.UpdateChannelRequest.meta.fields["channel"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["channel"].items():
@@ -9516,22 +9548,30 @@ def test_create_channel_connection_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(eventarc.CreateChannelConnectionRequest.meta.fields["channel_connection"].message, "DESCRIPTOR")
+    test_field = eventarc.CreateChannelConnectionRequest.meta.fields["channel_connection"]
 
-    if is_message_proto_plus_type:
-        message_fields = eventarc.CreateChannelConnectionRequest.meta.fields["channel_connection"].message.meta.fields
-    else:
-        message_fields = eventarc.CreateChannelConnectionRequest.meta.fields["channel_connection"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["channel_connection"].items():
@@ -10276,22 +10316,30 @@ def test_update_google_channel_config_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(eventarc.UpdateGoogleChannelConfigRequest.meta.fields["google_channel_config"].message, "DESCRIPTOR")
+    test_field = eventarc.UpdateGoogleChannelConfigRequest.meta.fields["google_channel_config"]
 
-    if is_message_proto_plus_type:
-        message_fields = eventarc.UpdateGoogleChannelConfigRequest.meta.fields["google_channel_config"].message.meta.fields
-    else:
-        message_fields = eventarc.UpdateGoogleChannelConfigRequest.meta.fields["google_channel_config"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["google_channel_config"].items():

--- a/tests/integration/goldens/redis/tests/unit/gapic/redis_v1/test_cloud_redis.py
+++ b/tests/integration/goldens/redis/tests/unit/gapic/redis_v1/test_cloud_redis.py
@@ -4307,22 +4307,30 @@ def test_create_instance_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(cloud_redis.CreateInstanceRequest.meta.fields["instance"].message, "DESCRIPTOR")
+    test_field = cloud_redis.CreateInstanceRequest.meta.fields["instance"]
 
-    if is_message_proto_plus_type:
-        message_fields = cloud_redis.CreateInstanceRequest.meta.fields["instance"].message.meta.fields
-    else:
-        message_fields = cloud_redis.CreateInstanceRequest.meta.fields["instance"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["instance"].items():
@@ -4604,22 +4612,30 @@ def test_update_instance_rest(request_type):
     # See https://github.com/googleapis/gapic-generator-python/issues/1748
 
     # Determine if the message type is proto-plus or protobuf
-    is_message_proto_plus_type = not hasattr(cloud_redis.UpdateInstanceRequest.meta.fields["instance"].message, "DESCRIPTOR")
+    test_field = cloud_redis.UpdateInstanceRequest.meta.fields["instance"]
 
-    if is_message_proto_plus_type:
-        message_fields = cloud_redis.UpdateInstanceRequest.meta.fields["instance"].message.meta.fields
-    else:
-        message_fields = cloud_redis.UpdateInstanceRequest.meta.fields["instance"].message.DESCRIPTOR.fields
+    def get_message_fields(field):
+        # Given a field which is a message (composite type), return a list with
+        # all the fields of the message.
+        # If the field is not a composite type, return an empty list.
+        message_fields = []
+
+        if hasattr(field, "message") and field.message:
+            is_field_type_proto_plus_type = not hasattr(field.message, "DESCRIPTOR")
+
+            if is_field_type_proto_plus_type:
+                message_fields = field.message.meta.fields.values()
+            else:
+                message_fields = field.message.DESCRIPTOR.fields
+        return message_fields
+
+    runtime_nested_fields = [
+        (field.name, nested_field.name)
+        for field in get_message_fields(test_field)
+        for nested_field in get_message_fields(field)
+    ]
 
     subfields_not_in_runtime = []
-
-    # Get all subfields for the message
-    runtime_nested_fields = [
-        (field.name, subfield.name)
-        for field in message_fields
-        if hasattr(field, "message_type") and field.message_type
-        for subfield in field.message_type.fields
-    ]
 
     # For each item in the sample request, create a list of sub fields which are not present at runtime
     for field, value in request_init["instance"].items():

